### PR TITLE
UI Builder: Add support to Fabric Components

### PR DESCRIPTION
**Pull request by vyhnalekl**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Added support to Fabric Buttons, ('FabricButton', 'FabricDefaultButton', 'FabricActionButton'). Added property to `JSONTreeItem` to differentiate the components - `moduleName`. Added `StyleProvider` which is fixing the issue with CSS not being available to fabric components inside of an iframe.

### CODE REVIEW

File: packages/fluentui/react-builder/src/config.tsx [hunk 1]
```diff
@@ -37,9 +38,17 @@ export const COMPONENT_GROUP = {
   Forms: ['Input', 'Dropdown', 'Form', 'Checkbox', 'RadioGroup', 'Slider', 'TextArea'],
   Actionable: ['Button', 'MenuButton', 'SplitButton', 'Menu', 'Toolbar'],
   Containers: ['Card', 'Carousel', 'Accordion', 'Segment', 'List', 'Tree', 'HierarchicalTree'],
+  Fabric: ['FabricButton', 'FabricDefaultButton', 'FabricActionButton'],
```
Comment (mnajdova) [inline packages/fluentui/react-builder/src/config.tsx:41]:
```suggestion
  FabricActionable: ['FabricButton', 'FabricDefaultButton', 'FabricActionButton'],
```

File: packages/fluentui/react-builder/src/utils/StyleProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+export const useProjectedStyles = (childWindow: Window, tree: JSONTreeElement) => {
+  React.useEffect(() => {
+    return childWindow ? projectStyles(childWindow) : undefined;
+  }, [childWindow, tree]);
+};
+
+export const StyleProvider: React.FunctionComponent<{ childWindow: Window; tree: JSONTreeElement }> = props => {
+  console.log('styleprovider', props.childWindow);
```
Comment (mnajdova) [inline packages/fluentui/react-builder/src/utils/StyleProvider.tsx:104]:
```suggestion
```

File: packages/fluentui/react-builder/src/utils/StyleProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  window.document.head.querySelectorAll('[data-merge-styles]').forEach((sourceStyleElement: HTMLStyleElement) => {
+    const { cssRules: sourceCssRules } = sourceStyleElement.sheet as CSSStyleSheet;
+
+    Array.from(sourceCssRules).forEach(rule => {
+      targetSheet.insertRule(rule.cssText, targetSheet.cssRules.length);
+    });
+  });
+
+  const dispose = () => {
+    console.log('disposing projecting styles', childWindow);
```
Comment (mnajdova) [inline packages/fluentui/react-builder/src/utils/StyleProvider.tsx:86]:
```suggestion
```

File: packages/fluentui/react-builder/src/utils/StyleProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+/**
+ * Manages tracking the child window and pushing styling changes from the main window to the child.
+ * When the child unloads, we will remove references appropriately avoiding memory leaks.
+ * @param childWindow Child window to project parent window styling to.
+ * @returns A dispose function for manually disposing; will disconnect on child window unload as well.
+ */
+export function projectStyles(childWindow: Window): () => void {
+  console.log('projecting styles', childWindow);
```
Comment (mnajdova) [inline packages/fluentui/react-builder/src/utils/StyleProvider.tsx:59]:
```suggestion
```

**Comment by ecraig12345** [comment]
@vyhnalekl Guessing this is no longer needed?

**Comment by levithomason** [comment]
@miroslavstastny will be helping us coordinate internally on merging or closing this up.

**Comment by ecraig12345** [comment]
Closing since this seems unlikely to be relevant at this point

**Pull request closed without merging**